engine/src/main/java/engine/player/command/PlayerCommandProcessor.java

```diff
@@ -8,24 +8,36 @@
 
 import java.util.Collection;
 import java.util.HashSet;
+import java.util.concurrent.BlockingQueue;
 
 public class PlayerCommandProcessor {
 
+    private static final int RETRIEVE_AMOUNT_PER_TERN = 100;
+
     private final PlayerManager playerManager;
 
     public PlayerCommandProcessor(PlayerManager playerManager) {
         this.playerManager = playerManager;
     }
 
-    public Collection<EngineCommand> processPlayerCommands(Collection<PlayerCommand> playerCommands) {
+    public Collection<EngineCommand> processPlayerCommands(BlockingQueue<PlayerCommand> playerCommands) {
         Collection<EngineCommand> engineCommands = new HashSet<>();
 
-        for (PlayerCommand playerCommand : playerCommands) {
+        int i = 0;
+
+        while (playerCommands.size() > 0 && i < RETRIEVE_AMOUNT_PER_TERN) {
```

Comment:
RETRIEVE_AMOUNT_PER_UPDATE*

Reply:
what about this method?
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html#drainTo(java.util.Collection,%20int)